main.py

```diff
@@ -1,11 +1,126 @@
 import argparse
+import logging
+import time
+from io import BytesIO
+from PIL import Image
 import logger
 import sound as snd
+import pygame.image
+import camera
 import tcp
-
+import udp
+import struct
 from threading import Thread
 
 buffer_size = None
+INFORMATION_SIZE = struct.calcsize("i" * 3)
+CHUNK_SIZE = 1500 - INFORMATION_SIZE
+
+
+def get_and_send_data(sock, resolution=(640, 480)):
+    sock.settimeout(100)
+    cam = camera.camera_init()
+
+    current_image_number = 0
+    try:
+        while True:
+            while True:
+                if cam.query_image():
+                    camera_image = cam.get_image()
+                    break
+
+            buffer = BytesIO()
+            im = Image.frombuffer("RGB", resolution, bytes(pygame.image.tostring(camera_image, "RGB")))
+            # NOTE: quality parameter was chosen experimentally
+            im.save(buffer, optimize=True, quality=45, format='JPEG')
+            image = buffer.getvalue()
+            buffer.close()
+
+            # To prevent integer overflow in bytes sending
+            if current_image_number == 2 ** 32 - 1:
+                current_image_number = 0
+
+            for i in range(len(image) // CHUNK_SIZE + (len(image) / CHUNK_SIZE != 0)):
```

Comment:
I think there's a bug.

I assume `len(image)` is never zero, so the second part `(len(image) / CHUNK_SIZE != 0)` is always `True`, so you can just rewrite it like that:

```suggestion
            for i in range(len(image) // CHUNK_SIZE + 1):
```

But! I think it's not actually what you wanted to do here. I assume it was something like:

```suggestion
            for i in range(int(math.ceil(len(image) / CHUNK_SIZE))):
```

Please double-check this.

Reply:
Yep, here supposed to be `%`, but i like the second variant more.